src/lib/pipeline/ReorderBuffer.cc

```diff
@@ -169,7 +169,8 @@ void ReorderBuffer::flush(uint64_t afterSeqId) {
     auto destinations = uop->getDestinationRegisters();
     for (int i = destinations.size() - 1; i >= 0; i--) {
       const auto& reg = destinations[i];
-      rat_.rewind(reg);
+      // Only rewind the register if it was renamed
+      if (reg.renamed) rat_.rewind(reg);
```

Comment:
Been thinking about this. We could move all this logic into the RAT? I feel that the register struct and the process of renaming should be separate. A register just sees it's self as a physical register so we could block a rewind if it has not been renamed instead?

Reply:
Not 100% sure of an implementation for this but will investigate if it's of interest

If it's not then to answer the original question, the proposed assertion would be good here

Reply:
Whilst I agree with the sentiment of keeping this inside of the RAT, I'm not sure of a way to do this nicely. Whereas the current implementation is very lightweight, and defaulting to `false` means it can remain un-used in other circumstances.

Personally I think the assertion is sufficient, but if you can think of a clean RAT-based implementation for this then that would be great

Reply:
Can find a clean solution, implemented assert